### PR TITLE
Merge `cardano-wallet`, `cardano-wallet-core-integration`, `cardano-wallet-cli` into `cardano-wallet-core`

### PR DESCRIPTION
**Pull request by Unisay**
- [x] Merge `cardano-wallet`, `cardano-wallet-core-integration`, `cardano-wallet-cli` into `cardano-wallet-core`
- [x] Update nix files
- [x] Regenerate nix files
- [x] Rename `lib/core` to `lib/wallet` with all the occurrences. 
- [x] Drop obsolete `nix/materialized/stack-nix`

### Issue Number

ADP-2225

### CODE REVIEW

File: lib/core/src/Cardano/Wallet/Shelley/Launch/Cluster.hs [hunk 1]
```diff
@@ -1049,49 +1049,49 @@ withCluster tr dir LocalClusterConfig{..} faucetFunds onClusterStart = bracketTr
...
-                -- setup working 100% correctly in alonzo will soon not be
-                -- important.
-                mapM_ (`registerViaTx` runningBFTNode) configuredPools
-                launchPools
-                    configuredPools
-                    genesisFiles
-                    (tail ports)
-                    runningBFTNode
-                    onClusterStart
+            then do
```
Comment (Anviking) [inline lib/core/src/Cardano/Wallet/Shelley/Launch/Cluster.hs]: The previous indentation was intentional to respect the 80-char line length. Why change it?

Comment (Unisay) [reply lib/core/src/Cardano/Wallet/Shelley/Launch/Cluster.hs]: I was fixing compilation errors similar to this one:
```
src/Cardano/Wallet/Shelley/Launch/Cluster.hs:1176:12: error:
    Unexpected semi-colons in conditional:
        if length (filter isRight group) /= poolCount; then do cancelAll
                                                               let errors
                                                                     = show (filter isLeft group)
                                                               throwIO
                                                                 $ ProcessHasExited
                                                                     ("cluster didn't start correctly: "
                                                                        <> errors)
                                                                     (ExitFailure
                                                                        1); else do let node = ...
                                                                                    action node
                                                                                      `finally`
                                                                                        cancelAll
    Perhaps you meant to use DoAndIfThenElse?
     |
1176 |         if length (filter isRight group) /= poolCount
     |     
```
... had to indent `then/else` relative to `if`.
Also, I made sure that 80 chars is not violated.

**Comment by Anviking** [comment]
There's a small number of  "lib/shelley" occurrences left:
```bash
git grep "lib/shelley"                                                                                                                                      1
.gitignore:lib/shelley/test/data/balanceTx/**/actual
docs/developers/Building.md:haskell-language-server lib/shelley/exe/cardano-wallet.hs
hie-direnv.yaml:# $ haskell-language-server lib/shelley/exe/cardano-wallet.hs
lib/wallet/test/data/cardano-node-shelley/start.sh:config_dir=lib/shelley/test/data/cardano-node-shelley
```

File: lib/wallet/cardano-wallet.cabal [hunk 2]
```diff
@@ -321,34 +384,200 @@ library
...
+  exposed-modules:
+      Test.Integration.Faucet
+      Test.Integration.Framework.Context
+      Test.Integration.Framework.DSL
+      Test.Integration.Framework.Request
+      Test.Integration.Framework.TestData
+      Test.Integration.Plutus
+      Test.Integration.Scenario.API.Blocks
+      Test.Integration.Scenario.API.Byron.Wallets
+      Test.Integration.Scenario.API.Byron.HWWallets
```
Comment (Anviking) [inline lib/wallet/cardano-wallet.cabal:471]: I guess the tests themselves could go in `cardano-wallet:integration` **eventually**. The benchmarks should only need setup and helpers like `Context` and `DSL`.

File: lib/wallet/src/Cardano/Pool/DB/Sqlite/TH.hs [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {-# LANGUAGE DataKinds #-}
 {-# LANGUAGE DeriveGeneric #-}
 {-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE EmptyDataDecls #-}
```
Comment (Anviking) [inline lib/wallet/src/Cardano/Pool/DB/Sqlite/TH.hs:4]: This was needed? (Btw, I didn't get: why did the merging cause some code changes to be needed? The default extensions are the same...?)

File: lib/core/src/Cardano/Wallet/Shelley/Launch/Cluster.hs (hunk 1 above)
Comment (Anviking) [reply lib/core/src/Cardano/Wallet/Shelley/Launch/Cluster.hs]: I tried de-indenting `then/else` here, and `cabal test cardano-wallet:integration` compiled fine. Are you sure you get these errors 🤔

File: lib/wallet/src/Cardano/Pool/DB/Sqlite/TH.hs (hunk 3 above)
Comment (Anviking) [reply lib/wallet/src/Cardano/Pool/DB/Sqlite/TH.hs:4]: Related: https://github.com/input-output-hk/cardano-wallet/pull/3505#discussion_r975279343

Comment (Unisay) [reply lib/wallet/src/Cardano/Pool/DB/Sqlite/TH.hs:4]: I think that I accidentally removed (and later restored) the `Haskell2010` language in the cabal file. That led to some compiler errors that I fixed exactly how compiler asked to fix them. Later on I realized the language issue and decided not to rollback my fixes. We're `Haskell98` compatible now :)

File: lib/wallet/cardano-wallet.cabal (hunk 2 above)
Comment (Unisay) [reply lib/wallet/cardano-wallet.cabal:471]: Agree, there is concern separation work to be done for integration tests. Its just that it doesn't make sense for it to delay merge.

File: nix/haskell.nix [hunk 4]
```diff
@@ -204,20 +204,6 @@ haskell-nix: haskell-nix.cabalProject' [
...
-                # Attempt to ensure visible progress in the macOS hydra job.
-                #
-                # An hypothesis is that #2472 is caused by heavy load and unfocused
-                # resources from running the tests concurrently, risking that the slowest
-                # hspec runner - and therefore the stdout - being silent for 900s causing
-                # hydra to timeout.
-                #
-                # Setting -j 1 should hopefully focus the resource we have in one place. It
-                # should go silent less often, at the expense of the full run getting slower.
-                unit.testFlags = lib.optionals pkgs.stdenv.hostPlatform.isDarwin [ "-j" "1" ];
```
Comment (Anviking) [inline nix/haskell.nix:218]: Right, we don't require Darwin to succeed anymore in bors, so not moving this setting should be fine 👍

File: nix/windows-testing-bundle.nix [hunk 5]
```diff
@@ -17,9 +17,7 @@
 
 let
   testData = {
-    core = ../lib/core/test/data;
-    shelley = ../lib/shelley/test/data;
-    cli = ../lib/cli/test/data;
+    wallet = ../lib/wallet/test/data;
```
Comment (Anviking) [inline nix/windows-testing-bundle.nix:20]: Neater :)

**Review by Anviking — APPROVED** [review]
Thanks! lgtm!

**Comment by Unisay** [comment]
bors merge

**Pull request merged**